tests/drivers/flash_api/src/main.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021-2024 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/device.h>
+
+/** Test instrumentation **/
+/*
+ * Below structure gathers variables that may be used by simulate API calls to mock behaviour
+ * of a flash device; the variables may be used however it is desired, because the it is
+ * up to mocked function and check afterwards to relate a variable to result
+ */
+static struct {
+	/* Set to value returned from any API call */
+	int ret;
+	/* Some size */
+	uint64_t size;
+	/* Allowed offset alignment */
+} simulated_values = {
+	.ret = 0,
+	.size = 0,
+};
+
+/*** Device definition atd == API Test Dev ***/
+static int some_get_size(const struct device *dev, uint64_t *size)
+{
+	__ASSERT_NO_MSG(dev != NULL);
+
+	*size = simulated_values.size;
+
+	return 0;
+}
+
+static int enotusp_get_size(const struct device *dev, uint64_t *size)
```

Comment:
```suggestion
static int enotsup_get_size(const struct device *dev, uint64_t *size)
```
?

Reply:
Eh, seems my brain was struggling between "enotsup" and eunsup", there was no clear winner. Will fix.